### ISSUE TITLE
Godot ignores changes to C# files in modules/mono/glue/cs_files/ unless you remove ~/.local/share/godot/mono

### ISSUE THREAD

**Issue by aaronfranke** [godotengine/godot#17124]
**Godot version:** Latest master as of reporting time, 3d03bcd0636aacfe530c52fc1cdc32c2edffea04

**OS/device including version:** Ubuntu 16.04

**Issue description:** I am trying to change the files in `modules/mono/glue/cs_files/` but whatever changes I make do not show up in Godot.

**Steps to reproduce:** 

1. Download a fresh copy of Godot master from GitHub

2. Make any changes to files in cs_files. Or just straight up delete them. I deleted Vector3.cs.

3. Compile Godot with Mono, generate glue, compile again.

4. Open Godot, open a C# script, and use Vector3.

5. Notice that even with `modules/mono/glue/cs_files/Vector3.cs` completely deleted, Godot still compiles and runs perfectly fine! What‽

It's as if Godot is completely ignoring files in this folder. It does, unless you remove `~/.local/share/godot/mono`, regenerate glue, and recompile. That's not ideal.

**Comment by aaronfranke** [comment]
You can literally delete every file in `modules/mono/glue/cs_files` except for VERSION.txt and Godot will still compile and run fine.

**Comment by NathanWarden** [comment]
This is a little bit tricky. You need to also delete the solutions from .local/share/godot/mono also. I usually just delete the mono folder.

**Comment by neikeq** [comment]
You need to increment the version in VERSION.txt

**Comment by aaronfranke** [comment]
Incrementing the version did not seem to work, but deleting the mono folder in `~/.local/share/godot` worked. Anyway, now that I can compile and make changes to Godot, I can start contributing :)

I do have some concerns about this.

1. If I change these files, and commit back to master, then won't anyone else who compiles also have to delete `~/.local/share/godot/mono` to see my changes?

2. Do I have to delete, recompile, and regenerate glue every time I make a change to these files?

**Comment by Vega-KH** [comment]
OMG, a while back I tried to make some changes to CS files and kept compiling with no results.  I didn't think of this and was banging my head against the wall.  I thought I was just stupid.  I guess I was, come to think of it.

**Comment by NathanWarden** [comment]
@Vega-KH You aren't alone, haha. The same exact thing happened to me the first time I tried editing those files :)

**Comment by aaronfranke** [comment]
Suggested labels: bug, topic:mono, no milestone. 

The Godot compiler ignoring changes to source files is not ideal. However it is easy to work around, therefore it is low-priority. Perhaps we could solve this by checking the hash of the files, and recompile them if the files are different. On a related note, I think it would be a good idea to move glue generation functionality into the compile process. This would allow us to fix this issue and also make compiling Mono builds much easier.

**Comment by akien-mga** [comment]
What's the status on this issue?

**Comment by aaronfranke** [comment]
Godot no longer ignores changes. Rebuilding without deleting now includes any changes I make.

TL;DR The problem has been fixed, thanks @neikeq
